modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/FilterToSqlHelper.java

```diff
@@ -930,6 +934,20 @@ private String getPrimaryKeyColumnsAsCommaSeparatedList(
         return sb.toString();
     }
 
+    public Object visit(FilterFunction_arrayAnyMatch filter, Object extraData) {
+        Expression array = getParameter(filter, 0, true);
```

Comment:
Defensive programming needed here to fit in with the Filter flexibility in representation of values.

Reply:
Can you explain more precisely what you would do here?